File: app/backend/src/tv_watchlist/models/category_detail.py
"""Full category payload: schema, rows, and companion sheets."""

from __future__ import annotations

from tv_watchlist.models.category_summary import CategorySummary
from tv_watchlist.models.column_spec import ColumnSpec
from tv_watchlist.models.reference_sheet import ReferenceSheet
from tv_watchlist.models.watch_row import WatchRow


class CategoryDetail(CategorySummary):
    """A category plus its editable grid and read-only reference sheets."""

    columns: list[ColumnSpec]
    rows: list[WatchRow]
    reference_sheets: list[ReferenceSheet]

File: app/frontend/src/types/CategoryDetail.ts
import type { CategorySummary } from "./CategorySummary";
import type { ColumnSpec } from "./ColumnSpec";
import type { ReferenceSheet } from "./ReferenceSheet";
import type { WatchRow } from "./WatchRow";

export interface CategoryDetail extends CategorySummary {
  columns: ColumnSpec[];
  rows: WatchRow[];
  reference_sheets: ReferenceSheet[];
}
